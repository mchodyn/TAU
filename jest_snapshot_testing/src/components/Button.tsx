import React from 'react';
import './Button.css'

interface buttonProps {
    name:string;
}

export const Button:React.FC<buttonProps> = props => {
    return (<button onClick={()=> window.alert(`Hello ${props.name}`)}>Changed text!</button>)
}

