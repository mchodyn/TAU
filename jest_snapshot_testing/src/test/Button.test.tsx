import React from 'react';
import {Button} from "../components/Button";
import renderer from 'react-test-renderer';

test('snapshot test', () => {
    const tree = renderer.create(<Button name={'Stranger'}/>).toJSON();
    expect(tree).toMatchSnapshot();
});
