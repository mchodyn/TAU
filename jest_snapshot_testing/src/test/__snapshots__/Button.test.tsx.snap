// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<button
  onClick={[Function]}
>
  Changed text!
</button>
`;
